config/routes.rb

```diff
@@ -1,4 +1,14 @@
 Rails.application.routes.draw do
+  root 'application#index'
+
+  patch 'movies/:id/upvote' => 'movies#upvote', as: :upvote_movie
```

Comment:
These are good, however since we're using `resources` lines below, we can actually include these extra non-CRUD actions into our resources definitions:
```ruby
resources :albums do
  patch 'upvote'
end
```
This will give you a route to `albums/:id/upvote` which goes to `upvote` on `AlbumsController`, and automatically has the `album_upvote_path` route helper method.